pkg/controller/mongodb/mongodb_tls.go

```diff
@@ -0,0 +1,144 @@
+package mongodb
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/container"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/podtemplatespec"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/statefulset"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	mdbv1 "github.com/mongodb/mongodb-kubernetes-operator/pkg/apis/mongodb/v1"
+)
+
+type tlsValidationResult struct {
+	Valid   bool
+	Message string
+}
+
+// validateTLSConfig will check that the configured ConfigMap and Secret exist and that they have the correct fields.
+// If TLS is correctly configured, the function will return a result where result.Valid = true.
+// If it's incorrectly configured the result will instead have result.Valid = false and a message for the user in result.Message.
+func (r *ReplicaSetReconciler) validateTLSConfig(mdb mdbv1.MongoDB) (tlsValidationResult, error) {
+	if !mdb.Spec.Security.TLS.Enabled {
+		return tlsValidationResult{Valid: true}, nil
+	}
+
+	r.log.Info("Ensuring TLS is correctly configured")
+
+	// Ensure CA ConfigMap exists
+	var caConfigMap corev1.ConfigMap
+	if err := r.client.Get(context.TODO(), mdb.TLSConfigMapNamespacedName(), &caConfigMap); err != nil {
+		if errors.IsNotFound(err) {
+			return tlsValidationResult{
+				Valid:   false,
+				Message: fmt.Sprintf(`CA ConfigMap "%s" not found`, mdb.TLSConfigMapNamespacedName().String()),
+			}, nil
+		}
+
+		return tlsValidationResult{}, err
+	}
+
+	// Ensure ConfigMap has a "ca.crt" field
+	if cert, ok := caConfigMap.Data[tlsCACertName]; !ok || cert == "" {
+		message := fmt.Sprintf(`ConfigMap "%s" should have a CA certificate in field "%s"`, mdb.TLSConfigMapNamespacedName().String(), tlsCACertName)
+		return tlsValidationResult{Valid: false, Message: message}, nil
+	}
+
+	// Ensure Secret exists
+	var secret corev1.Secret
+	if err := r.client.Get(context.TODO(), mdb.TLSSecretNamespacedName(), &secret); err != nil {
+		if errors.IsNotFound(err) {
+			message := fmt.Sprintf(`Secret "%s" not found`, mdb.TLSSecretNamespacedName().String())
+			return tlsValidationResult{Valid: false, Message: message}, nil
+		}
+
+		return tlsValidationResult{}, err
+	}
+
+	// Ensure Secret has "tls.crt" and "tls.key" fields
+	if key, ok := secret.Data[tlsSecretKeyName]; !ok || len(key) == 0 {
+		message := fmt.Sprintf(`Secret "%s" should have a key in field "%s"`, mdb.TLSSecretNamespacedName().String(), tlsSecretKeyName)
```

Comment:
don't think we need the explicit `String()` call, this is handled by `Sprintf`

Reply:
Didn't know about that, fixed!